nmigen/lib/fifo.py

```diff
@@ -369,7 +369,7 @@ def elaborate(self, platform):
         produce_dec = m.submodules.produce_dec = \
             GrayDecoder(self._ctr_bits)
         m.d.comb += produce_dec.i.eq(produce_r_gry),
-        m.d[self._r_domain] += produce_r_bin.eq(produce_dec.o)
+        m.d.comb += produce_r_bin.eq(produce_dec.o)
```

Comment:
Wait, why does this PR change level-unrelated AsyncFIFO logic at all?

Reply:
Hmm... you are right, that seems wrong. 
I don't remember to write that, so we should probably ask @awygle or undo that change